compiler/modules/types/core.bal

```diff
@@ -649,6 +653,12 @@ public function intersect(SemType t1, SemType t2) returns SemType {
     return createComplexSemType(all, subtypes);    
 }
 
+public function intersectMemberSemTypes(Env env, MemberSemType t1, MemberSemType t2) returns MemberSemType {
```

Comment:
We were going to change this to 

- check if the MemberSemType contains anything from Cell
- if it does, do the special thing for Cells
- otherwise, do regular intersection

Reply:
I added these changes with integrating cells into list type PR. I'll add it here as well. :)

Reply:
Oh I see. Sorry for the confusion.